pkg/controllers/namespace/namespaceController.go

```diff
@@ -0,0 +1,268 @@
+				filteredProfiles = append(filteredProfiles, profile)
+			}
+		}
+		glog.Debugf("Found %d profiles in calico ETCD:", len(filteredProfiles))
+		return filteredProfiles, nil
+	}
+
+	// Function returns key of the object in kubernetes format.
+	keyFunc := func(obj interface{}) string {
+		profile := obj.(api.Profile)
+		Key := profile.Metadata.Name
+		return Key
+	}
+
+	cacheArgs := calicocache.ResourceCacheArgs{
+		ListFunc:   listFunc,
+		KeyFunc:    keyFunc,
+		Client:     cclient,
+		ObjectType: "*api.Profile", // Restrict cache to store pointers to calico profile objects only.
+	}
+
+	ccache := calicocache.NewResourceCache(cacheArgs)
+	namespaceConverter := converter.NewNamespaceConverter()
+
+	// create the watcher
+	listWatcher := cache.NewListWatchFromClient(clientset.Core().RESTClient(), "namespaces", "", fields.Everything())
+
+	// Bind the calico cache to kubernetes cache with the help of an informer. This way we make sure that
+	// whenever the kubernetes cache is updated, changes get reflected in calico cache as well.
+	indexer, informer := cache.NewIndexerInformer(listWatcher, &v1.Namespace{}, 0, cache.ResourceEventHandlerFuncs{
+		AddFunc: func(obj interface{}) {
+
+			key, err := cache.MetaNamespaceKeyFunc(obj)
+			glog.Infof("Got ADD event for namespace: %s\n", key)
+
+			if err == nil {
+				profile, conversionErr := namespaceConverter.Convert(obj)
+				if conversionErr != nil {
+					glog.Errorf("Error while converting %#v to calico profile.", obj)
+					return
+				}
+				// Add profileName:*profile in calicoCache
+				ccache.Set(profile.(*api.Profile).Metadata.Name, profile)
+			}
+		},
+		UpdateFunc: func(oldObj interface{}, newObj interface{}) {
+
+			key, err := cache.MetaNamespaceKeyFunc(newObj)
+
+			glog.Infof("Got UPDATE event for namespace: %s\n", key)
+			glog.Debugf("Old object: %#v\n", oldObj)
+			glog.Debugf("New object: %#v\n", newObj)
+
+			if err == nil {
+
+				if newObj.(*v1.Namespace).Status.Phase == "Terminating" {
+					// If object status is updated to "Terminating", object
+					// is getting deleted. Ignore this event. When deletion
+					// completes another DELETE event will be raised.
+					// Let DeleteFunc handle that.
+					glog.Debug("Namespace %s is getting deleted.", newObj.(*v1.Namespace).ObjectMeta.GetName())
+					return
+				}
+				profile, conversionErr := namespaceConverter.Convert(newObj)
+				if conversionErr != nil {
+					glog.Errorf("Error while converting %#v to calico profile.", newObj)
+					return
+				}
+				// Add profileName:profile in calicoCache
+				ccache.Set(profile.(*api.Profile).Metadata.Name, profile)
+			}
+		},
+		DeleteFunc: func(obj interface{}) {
+			// IndexerInformer uses a delta queue, therefore for deletes we have to use this
+			// key function.
+
+			key, err := cache.DeletionHandlingMetaNamespaceKeyFunc(obj)
+			glog.Infof("Got DELETE event for namespace: %s\n", key)
+
+			if err == nil {
+				calicoKey := fmt.Sprintf(converter.ProfileNameFormat+"%s", key)
```

Comment:
Calico cache is storing calico objects and the keys are their names which are in calico format (ns.projectcalico.org/test etc.) So, I think we can convert the object and then use its name